torch/csrc/jit/script/init.cpp

```diff
@@ -501,17 +507,20 @@ struct VISIBILITY_HIDDEN BooleanDispatchValue : public SugaredValue {
 };
 
 std::shared_ptr<MethodValue> moduleToMethod(
-    const std::shared_ptr<Module>& mod) {
+    const std::shared_ptr<Module>& mod,
+    const py::object& py_mod,
+    const SourceRange& loc) {
   // this path only supports calling raw script functions
   // but because they are not distinguished from models, we have to check
   // that they are function-like here. They must not have state, and they
   // must have a forward method. When we expose functions to python
   //  this will be replaced with a direct py::isinstance<Function> call.
 
   if (mod->get_parameters().size() != 0) {
-    throw ErrorReport()
-        << "Attempted to inline a Module with parameters. "
-           "Stateful modules to be inlined must be submodules of the callee.";
+    throw ErrorReport() << "Modules with parameters must be submodules of"
+                           " the caller, but this module was not: "
+                        << py::str(py_mod) << "\n"
```

Comment:
Could you post what the new error message looks like ? it's hard to tell from the diff

Reply:
It just adds a printout of what the module is (e.g. `Linear(10, 10)`) and the `SourceLocation` of the failing call (which should have been there in the first place)